src/app/pages/services/components/service-snmp/service-snmp.component.ts

```diff
@@ -1,177 +1,120 @@
-        }],
-    },
-    {
-      name: helptext.other_title,
-      width: '100%',
-      label: true,
-      config: [
-        {
-          type: 'textarea',
-          name: 'options',
-          placeholder: helptext.options_placeholder,
-          tooltip: helptext.options_tooltip,
-        },
-        {
-          type: 'checkbox',
-          name: 'zilstat',
-          placeholder: helptext.zilstat_placeholder,
-          tooltip: helptext.zilstat_tooltip,
-        },
-        {
-          type: 'select',
-          name: 'loglevel',
-          placeholder: helptext.loglevel_placeholder,
-          tooltip: helptext.loglevel_tooltip,
-          options: helptext.loglevel_options,
-        },
-        {
-          type: 'checkbox',
-          name: 'iftop',
-          placeholder: helptext.iftop.placeholder,
-          tooltip: helptext.iftop.tooltip,
-          value: false,
-        },
-      ],
-    },
-    { name: 'divider', divider: true },
-  ];
+export class ServiceSnmpComponent implements OnInit {
+  isFormLoading = false;
 
-  constructor(
-    protected router: Router,
-    protected route: ActivatedRoute,
-    protected ws: WebSocketService,
-    protected iscsiService: IscsiService,
-    protected idmapService: IdmapService,
-  ) {}
+  form = this.fb.group({
+    location: [''],
+    contact: ['', Validators.email],
+    community: ['', Validators.pattern(/^[\w\_\-\.\s]*$/)],
+
+    v3: [false],
+    v3_username: [''],
+    v3_authtype: [''],
+    v3_password: ['', [
+      Validators.minLength(8),
+      this.validation.validateOnCondition(
+        () => this.isV3SupportEnabled,
+        Validators.required,
+      ),
+    ]],
+    v3_privproto: [''],
+    v3_privpassphrase: ['', Validators.minLength(8)],
+
+    options: [''],
+    zilstat: [false],
+    loglevel: [null as number],
+    iftop: [false],
+  });
 
-  afterInit(entityForm: EntityFormComponent): void {
-    entityForm.submitFunction = this.submitFunction;
+  readonly tooltips = {
+    location: helptext.location_tooltip,
+    contact: helptext.contact_tooltip,
+    community: helptext.community_tooltip,
+    v3: helptext.v3_tooltip,
+    v3_username: helptext.v3_username_tooltip,
+    v3_authtype: helptext.v3_authtype_tooltip,
+    v3_password: helptext.v3_password_tooltip,
+    v3_privproto: helptext.v3_privproto_tooltip,
+    v3_privpassphrase: helptext.v3_privpassphrase_tooltip,
+    options: helptext.options_tooltip,
+    zilstat: helptext.zilstat_tooltip,
+    loglevel: helptext.loglevel_tooltip,
+    iftop: helptext.iftop.tooltip,
+  };
+
+  readonly authtypeOptions$ = of(helptext.v3_authtype_options);
+  readonly privprotoOptions$ = of(helptext.v3_privproto_options);
+  readonly logLevelOptions$ = of(helptext.loglevel_options);
+
+  get isV3SupportEnabled(): boolean {
+    return this.form?.value?.['v3'];
   }
 
-  resourceTransformIncomingRestData(data: SnmpConfig): any {
-    delete data['v3_privpassphrase'];
```

Comment:
We were not filling in old passwords and passphrases fetched from the API upon edit.

Reply:
Yeah, weird. Don't see why not though.